etl/controller/pulsemaintainer.go

```diff
@@ -99,9 +102,81 @@ func (c *Controller) pulseSequence(ctx context.Context) {
 }
 
 func pulseIsComplete(p types.Pulse, d []string) bool { // nolint
-	// TODO implement me
-	// This if is here for test reason, delete it after implementation and update test data for expected behavior
-	return p.PulseNo >= 0
+	if len(d) == 0 {
+		return false
+	}
+
+	// root
+	if len(d) == 1 && d[0] == "" {
+		return true
+	}
+
+	// inverts last bool symbol in string
+	invertLastSymbol := func(s string) string {
+		last := s[len(s)-1:]
+		var invertedSymbol string
+		b, err := strconv.ParseBool(last)
+		if err != nil {
+			log.Fatal("non binary symbol in jet id: ", last)
+		}
+
+		if b {
+			invertedSymbol = "0"
+		} else {
+			invertedSymbol = "1"
+		}
+
+		return s[:len(s)-1] + invertedSymbol
+	}
+
+	// making map for easy search
```

Comment:
You can change list to map at function params (and in `jetDropRegister`), if it more suitable

Reply:
seems like too much code on adding to map in SetJetDropData, than adding to slice
let's leave it as it is